src/coreComponents/meshUtilities/InternalWellboreGenerator.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file InternalWellboreGenerator.cpp
+ */
+
+#include "InternalWellboreGenerator.hpp"
+#include "managers/DomainPartition.hpp"
+#include "mpiCommunications/PartitionBase.hpp"
+
+namespace geosx
+{
+using namespace dataRepository;
+
+InternalWellboreGenerator::InternalWellboreGenerator( string const & name, Group * const parent ):
+  InternalMeshGenerator( name, parent )
+{}
+
+void InternalWellboreGenerator::generateMesh( DomainPartition & domain )
+{
+  InternalMeshGenerator::generateMesh( domain );
+
+  Group & meshBodies = domain.getGroup( string( "MeshBodies" ));
+  MeshBody & meshBody = meshBodies.registerGroup< MeshBody >( this->getName() );
+  MeshLevel & meshLevel0 = meshBody.registerGroup< MeshLevel >( string( "Level0" ));
+  NodeManager & nodeManager = meshLevel0.getNodeManager();
+  Group & nodeSets = nodeManager.sets();
+
+  // Wellbore nodeset
+  SortedArray< localIndex > & rnegNodes = nodeSets.registerWrapper< SortedArray< localIndex > >( string( "rneg" ) ).reference();
+  SortedArray< localIndex > & rposNodes = nodeSets.registerWrapper< SortedArray< localIndex > >( string( "rpos" ) ).reference();
+  SortedArray< localIndex > & tnegNodes = nodeSets.registerWrapper< SortedArray< localIndex > >( string( "tneg" ) ).reference();
+  SortedArray< localIndex > & tposNodes = nodeSets.registerWrapper< SortedArray< localIndex > >( string( "tpos" ) ).reference();
+
+  arrayView2d< real64, nodes::REFERENCE_POSITION_USD > const & X = nodeManager.referencePosition();
+  for( int localNodeIndex=0; localNodeIndex<nodeManager.size(); ++localNodeIndex )
+  {
+    real64 xCoord = X( localNodeIndex, 0 );
+    real64 yCoord = X( localNodeIndex, 1 );
```

Comment:
I don't know but is there an enum somewhere for this `0` and `1`?

Plus you should define your variables `const`.

Reply:
No, 0 and 1 are just the first and second components of a node's position.
`const` is added

Reply:
ok then